src/algebraic_geometry/structure_sheaf.lean

```diff
@@ -777,4 +782,133 @@ def basic_open_iso (f : R) : (structure_sheaf R).presheaf.obj (op (basic_open f)
   CommRing.of (localization.away f) :=
 (as_iso (to_basic_open R f)).symm
 
+section comap
+
+variables {R} {S : Type u} [comm_ring S] {P : Type u} [comm_ring P]
+
+/--
+For a ring homomorphism `f : R →+* S` and open sets `U` and `V` of the prime spectra of `R` and
+`S` such that `V ⊆ (comap f) ⁻¹ U`, the induced ring homomorphism from the structure sheaf of `R`
+at `U` to the structure sheaf of `S` at `V`.
+
+If a section `s` on `U` is locally equal to the fraction `a/b`, its image on `V` is locally equal
+to the fraction `f(a)/f(b)`.
+-/
+@[simps] def structure_sheaf.comap (f : R →+* S) (U : opens (prime_spectrum.Top R))
```

Comment:
Does `@[simps]` generate a useful `_apply` lemma here? The `to_fun` entry looks pretty intimidating.

Reply:
Yes, it's used multiple times in the subsequent lemmas. The bulk of the `to_fun` entry is the proof that the section on `V` is again locally fraction, which doesn't show up in the `_apply` lemma any more. Still it's not exactly easy to read, I agree. I'm not sure how to improve it though. The comment above is meant to give an idea of what the definition means. Maybe I should expand upon that?

Reply:
One way to improve it might be to make an earlier definition `comap_fun` and use that here. That way you can govern to what extent `simp` will explode into a huge formula.

Reply:
Okay, I will try to do that!